akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala

```diff
@@ -86,4 +89,15 @@ class ScatterGatherFirstCompletedSpec extends AkkaSpec with DefaultTimeout with
 
   }
 
+  "Scatter-gather pool" must {
+
+    "without routees should reply immediately" in {
+      val probe = TestProbe()
+      val router = system.actorOf(ScatterGatherFirstCompletedPool(nrOfInstances = 0, within = 5.seconds).props(Props.empty))
```

Comment:
I thought that we may consider `0` illegal here, but the case Heiko spoke of can well be triggered by all the initially in-the-pool routees to die, leaving the pool having 0 routees... then the optimisation makes sense, right?